src/cpp/shared/combat/army.cpp

```diff
@@ -1790,4 +1791,9 @@ void army::InitClean() {
   this->mirrorIdx = -1;
   this->armyIdx = -1;
   this->previousQuantity = -1;
-}
\ No newline at end of file
+}
+
+void army::InitClean() {
+	this->InitClean_orig();
```

Comment:
There's no need to redefine InitClean_orig. You can just use the version in the assembly.